src/main/kotlin/org/rust/ide/intentions/ExpandModule.kt

```diff
@@ -17,7 +20,7 @@ class ExpandModule : IntentionAction {
     override fun startInWriteAction()   = true
 
     override fun isAvailable(project: Project, editor: Editor?, file: PsiFile?) =
-        file is RustFile && file.name != RustModules.MOD_RS
+        file is RustFile && (!(file.rustMod as RustModItem).ownsDirectory && !(file.name.equals(RustModules.LIB_RS) && file.containingDirectory?.name.equals("src")))
```

Comment:
The `|| isCrateRoot` condition in `ownsDirecotry` should handle `src/lib.rs`, `src/bin/my_executable.rs`, `examples/hello_world.rs` and all other possible locations of crate roots. 

It only works if you `import` project via cargo though, but this is a necessary limitation.